File: lib/spiral/async_.py
from collections.abc import Callable
from typing import Any
from fable_modules.fable_library.async_builder import (singleton, CancellationToken, Async)

def method0(v0_1: CancellationToken) -> Async[CancellationToken]:
    def _arrow100(__unit: None=None, v0_1: Any=v0_1) -> Async[CancellationToken]:
        return singleton.Return(None)

    return singleton.Delay(_arrow100)


def closure0(unit_var: None, v0_1: CancellationToken) -> Async[CancellationToken]:
    return method0(v0_1)


def _arrow101(v: CancellationToken) -> Async[CancellationToken]:
    return closure0(None, v)


v0: Callable[[CancellationToken], Async[CancellationToken]] = _arrow101

def merge_cancellation_token_with_default_async(x: CancellationToken) -> Async[CancellationToken]:
    return v0(x)

File: lib/spiral/async_.ts
import { Async, singleton } from "../../deps/Fable/src/fable-library-ts/AsyncBuilder.js";
import { defaultOf } from "../../deps/Fable/src/fable-library-ts/Util.js";

export function method0(v0_1: any): Async<any> {
    return singleton.Delay<any>((): Async<any> => {
        const v24: any = defaultOf();
        return singleton.Return<any>(v24);
    });
}

export function closure0(unitVar: void, v0_1: any): Async<any> {
    return method0(v0_1);
}

export const v0 = (v: any): Async<any> => closure0(undefined, v);

export function merge_cancellation_token_with_default_async(x: any): Async<any> {
    return v0(x);
}


